### DISCUSSION TITLE
sorting by frontmatter integers

### DISCUSSION THREAD

**Discussion by ahillio** [11ty/eleventy#2158] [Q&A] [answered]
I'm trying to do what's described at https://github.com/11ty/eleventy/issues/898, sorting by an `order` integer from frontmatter like:

```mkd
---
title: My awesome post
order: 1
---

Content
```

It seems to do that a custom function must created, and then that should be passed to `eleventyConfig.addFilter`...?  That's what people describe, and I tried to copy exactly what's suggested there.

I have this in my `.eleventy.js`:

```js
  function sortByOrder(values) {
    let vals = [...values];
    return vals.sort((a, b) => Math.sign(a.data.order - b.data.order));
  }
  eleventyConfig.addFilter('sortByOrder', sortByOrder);
```
but then in my liquid template, doing:

```liquid
{%- for post in collections.oikos sortByOrder -%}
```
results in: `illegal tag` parse errors.

I'm using 0.12.1, and sadly spending hours trying to sort :(

**Comment by pdehaan** [comment, marked as answer]
@ahillio I think you're missing a `|` before the `sortByOrder` filter.
This worked for me (using Nunjucks template engine):

```njk
{%- for post in collections.oikos | sortByOrder -%}
  {{ post.data.title }} ({{ post.data.order }})
{% endfor %}
```

**Reply by pdehaan** [reply]
Oh sorry, you said _liquid_ template. Syntax is slightly different. I think you need to assign the filtered collection to a variable first and THEN loop over it:

```liquid
{%- assign posts = collections.oikos | sortByOrder -%}
{%- for post in posts -%}
  {{ post.data.title }} ({{ post.data.order }})
{% endfor %}
```

**Reply by ahillio** [reply]
Ohmygoodness - THANK YOU <3

with exlamation marks too!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
